typegate/src/typegate/artifacts/mod.ts

```diff
@@ -20,27 +23,24 @@ function getUploadPath(tgName: string) {
 export async function getLocalPath(
   meta: ArtifactMeta,
   mainModuleMeta: ArtifactMeta,
+  dirs: Dirs,
 ) {
-  const cachedPath = resolve(STORE_DIR, meta.hash);
+  const cachedPath = resolve(dirs.cache, meta.hash);
   const localPath = resolve(
-    ARTIFACTS_DIR,
+    dirs.artifacts,
     mainModuleMeta.hash,
     meta.typegraphName,
     meta.relativePath,
   );
 
-  // TODO: what happens when symlink already exists? or when same local path artifacts with different cachedPath
-  try {
-    const fileInfo = await Deno.lstat(localPath);
-    if (fileInfo.isFile || fileInfo.isSymlink) {
-      await Deno.remove(localPath);
-    }
-    console.log(`Removed existing link on: ${localPath}`);
-  } catch {
-    // link doesn't exist before
+  if (await exists(localPath)) {
```

Comment:
what if it's a dependency with the same name(same `localPath`) but different content, shouldn't we overwrite the existing file?

Reply:
Good point.
I think we should change the parent dir (currently, entrypoint hash) to include all the hashes of the deps.

In the case you mentionned, Deno would use the cached data instead of reading from the file...